infer/modules/train/extract/extract_f0_print.py

```diff
@@ -1,175 +1,155 @@
 import os
 import sys
-import traceback
 
-import parselmouth
+import numpy as np
 
 now_dir = os.getcwd()
 sys.path.append(now_dir)
 import logging
 
-import numpy as np
-import pyworld
+logging.getLogger("numba").setLevel(logging.WARNING)
+logger = logging.getLogger("rvc.F0Print")
+
+from rvc.f0 import F0Predictor, CRePE, PM, Dio, Harvest, RMVPE, FCPE
 
 from infer.lib.audio import load_audio
 
-logging.getLogger("numba").setLevel(logging.WARNING)
-from multiprocessing import Process
-
-exp_dir = sys.argv[1]
-f = open("%s/extract_f0_feature.log" % exp_dir, "a+")
-
-
-def printt(strr):
-    print(strr)
-    f.write("%s\n" % strr)
-    f.flush()
-
-
-n_p = int(sys.argv[2])
-f0method = sys.argv[3]
-
-
-class FeatureInput(object):
-    def __init__(self, samplerate=16000, hop_size=160):
-        self.fs = samplerate
-        self.hop = hop_size
-
-        self.f0_bin = 256
-        self.f0_max = 1100.0
-        self.f0_min = 50.0
-        self.f0_mel_min = 1127 * np.log(1 + self.f0_min / 700)
-        self.f0_mel_max = 1127 * np.log(1 + self.f0_max / 700)
-
-    def compute_f0(self, path, f0_method):
-        x = load_audio(path, self.fs)
-        p_len = x.shape[0] // self.hop
-        if f0_method == "pm":
-            time_step = 160 / 16000 * 1000
-            f0_min = 50
-            f0_max = 1100
-            f0 = (
-                parselmouth.Sound(x, self.fs)
-                .to_pitch_ac(
-                    time_step=time_step / 1000,
-                    voicing_threshold=0.6,
-                    pitch_floor=f0_min,
-                    pitch_ceiling=f0_max,
-                )
-                .selected_array["frequency"]
-            )
-            pad_size = (p_len - len(f0) + 1) // 2
-            if pad_size > 0 or p_len - len(f0) - pad_size > 0:
-                f0 = np.pad(
-                    f0, [[pad_size, p_len - len(f0) - pad_size]], mode="constant"
-                )
-        elif f0_method == "harvest":
-            f0, t = pyworld.harvest(
-                x.astype(np.double),
-                fs=self.fs,

```

Comment:
You should not specify these values explicitly but use self.hop, self.f0_min, self.f0_max, etc.

Reply:
Shouldn't those be used automatically if no value is passed?

Reply:
> Shouldn't those be used automatically if no value is passed?

The automatic value is different from what currently used.

Reply:
> > Shouldn't those be used automatically if no value is passed?
> 
> The automatic value is different from what currently used.

So... what's the problem again? ._.
I'm sorry I'm just confused..

Reply:
You can see that in this original code, it used self.f0_max, self.f0_min, self.hop, etc. So you should also use that reference, not the absolute value.

Reply:
> You can see that in this original code, it used self.f0_max, self.f0_min, self.hop, etc. So you should also use that reference, not the absolute value.

Do you mean something like this?
```py
        predictor = predictor(
            hop_length=predictor.hop_length, f0_min=predictor.f0_min, f0_max=predictor.f0_max, sampling_rate=16000, device=device
        )
```
Because i don't have a class FeatureInput like in the original code, I have no `self` of reference to pass as the value

I'm not really sure why even pass parameters if they're going to be the default ones...

Reply:
OK. Well, the typical value should come from config.json but since the original code just specify it, you can specify it too. I will unify them later. Also, the hop_len is ALWAYS 160, don't use the value 512.

Reply:
> OK. Well, the typical value should come from config.json but since the original code just specify it, you can specify it too. I will unify them later. Also, the hop_len is ALWAYS 160, don't use the value 512.

Yeah, I got confused because your new implementation of F0Predictor uses 512 by default
Changed to 160 in 2e405c2 though!

Reply:
> I got confused because your new implementation of F0Predictor uses 512 by default

It is deliberately implemented to make sure all the parameters are filled in correctly. If not, the result will be incorrect.